### PR TITLE
Switch to codecov

### LINKED ISSUE #8: Use codecov to monitor code coverage
We are already using it in:

https://github.com/ZEISS/libczi
https://github.com/ZEISS/imgdoc2
The nice thing is that you get PR comments, dashboards and everything else as part of one solution stack.

https://app.codecov.io/gh/ZEISS/libczi

Also: We do not need to rely on more/other community actions.

### PR DESCRIPTION
**Pull request by FelixS90**
## Description

Use codecov to have fullstack code coverage support in PR, dashboard in one solution
No need for any other third party actions

Fixes #8 

**Note:** 
Status checks for branch policy need to be updated after merging this PR since job names have changed -> will take care about it.
Actions previously added to the allowed actions in the branch policy need to be removed again -> will take care about it.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
PR
https://app.codecov.io/gh/ZEISS/czicompress/tree/8-use-codecov-to-monitor-code-coverage
There is no component-based badge available yet, i.e. average coverage for czishrink and czicompress are displayed and badge. However, when clicking on badge, codecov dashboard shows everything.

## Checklist:

- [x] I followed the Contributing Guidelines.
- [x] I did a self-review.
- [ ] I commented my code, particularly in hard-to-understand areas.
- [ ] I updated the documentation.
- [ ] I updated the version of czicompress following [the README (Versioning)](../README.md#versioning) depending on the [type of change](#type-of-change)
  - Bug fix -> PATCH
  - New feature -> MINOR
  - Breaking change -> MAJOR
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

**Review by m-ringler — CHANGES_REQUESTED** [review]

File: .github/workflows/czishrink_dotnet.yml [hunk 1]
```diff
@@ -69,12 +69,12 @@ jobs:
         run: dotnet restore
 
       - name: Build
-        run: dotnet build --no-restore -c Release
+        run: dotnet build --no-restore -c Debug
```
[the remark this answers is not in the document]
Comment (FelixS90) [reply .github/workflows/czishrink_dotnet.yml:72]: Will fix.

File: .github/workflows/czicompress_cmake.yml [hunk 2]
```diff
@@ -81,9 +88,32 @@ jobs:
...
+        if: ${{ (matrix.config.os == 'windows-latest') && (matrix.build == 'Debug') }}
+        run: |
+          choco install OpenCppCoverage -y
+          echo "C:\Program Files\OpenCppCoverage" >> $env:GITHUB_PATH
+
+      - name: Get Coverage
+        if: ${{ (matrix.config.os == 'windows-latest') && (matrix.build == 'Debug') }}
+        working-directory: ${{github.workspace}}/czicompress/build/tests/${{matrix.build}}
+        shell: cmd
+        run: OpenCppCoverage.exe --export_type cobertura:${{github.workspace}}\TestResults\coverage.xml --config_file "${{github.workspace}}\opencppcoverage.txt" -- czicompress_tests.exe
```
[the remark this answers is not in the document]
Comment (FelixS90) [reply .github/workflows/czicompress_cmake.yml:92]: Will fix.

File: czicompress/README.md [hunk 3]
```diff
@@ -38,8 +38,8 @@ The tool is based on [libczi](https://github.com/ZEISS/libczi.git).
 ## Download
 We have not yet published any Releases. Meanwhile, you can download binaries from the artifacts of the latest [Build workflow run](https://github.com/ZEISS/czicompress/actions/workflows/czicompress_cmake.yml?query=branch%3Amain).
 Click on the topmost successful run and download the binaries for your platform:
-* czicompress-windows-64-release-msvc-package-off for Windows
-* czicompress-ubuntu-release-llvm-package-off for Linux
+* czicompress-windows-64-msvc-package-on-Release for Windows
+* czicompress-ubuntu-llvm-package-off-Release for Linux
```
Comment (m-ringler) [inline czicompress/README.md:42]: Shouldn't this be
```suggestion
* czicompress-ubuntu-package-off-Release for Linux
```

**Review by m-ringler — DISMISSED** [review]

File: czicompress/README.md (hunk 3 above)
Comment (FelixS90) [reply czicompress/README.md:42]: Good catch. Was obviously wrong before already. Will fix.

**Review by m-ringler — APPROVED** [review]

**Pull request merged**